File: src/mysqlcookbook/recipes/metadata/get_server_version.rb
#!/usr/bin/ruby
# get_server_version.rb: get server version string and number from server

# This script demonstrates how to get the string by issuing a
# SELECT VERSION() statement.

require "Cookbook"

# Return an array consisting of the server version number in both string
# and numeric forms.

def get_server_version(client)
  # fetch result into scalar string
  ver_str = client.query("SELECT VERSION()").first.values[0]
  major, minor, patch = ver_str.split(/\./)
  patch.sub!(/\D.*$/, "")  # strip nonnumeric suffix if present
  ver_num = major.to_i*10000 + minor.to_i*100 + patch.to_i
  [ver_str, ver_num]
end

begin
  client = Cookbook.connect
  # get server version string and number and print them
  ver_str, ver_num = get_server_version(client)
  puts "Version: #{ver_str} (#{ver_num})"
  client.close
rescue Mysql2::Error => e
  puts "Error code: #{e.errno}"
  puts "Error message: #{e.message}"
end

File: src/mysqlcookbook/recipes/metadata/get_server_version.py
#!/usr/bin/python3
# get_server_version.py: get server version string and number from server

# This script demonstrates how to get the string by issuing a
# SELECT VERSION() statement.

import re
import mysql.connector
import cookbook

# Return a list consisting of the server version number in both string
# and numeric forms.

def get_server_version(conn):
  # fetch result into scalar string
  cursor = conn.cursor()
  cursor.execute("SELECT VERSION()")
  ver_str = cursor.fetchall()[0][0]
  cursor.close()
  (major, minor, patch) = ver_str.split('.')
  patch = re.sub('\D.*$', "", patch)  # strip nonnumeric suffix if present
  ver_num = int(major)*10000 + int(minor)*100 + int(patch)
  return (ver_str, ver_num)

try:
  conn = cookbook.connect()
  # get server version string and number and print them
  (ver_str, ver_num) = get_server_version(conn)
  print("Version: %s (%d)" % (ver_str, ver_num))
  conn.close()
except mysql.connector.Error as e:
  print("Error code: %s" % e.errno)
  print("Error message: %s" % e.msg)
